### PR TITLE
Echo

### PR DESCRIPTION
**Pull request by henry808**

### CODE REVIEW

File: echo_client.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        if len(msg_part) < buffersize:
+            done = True
+        text_back += msg_part
+    client_socket.close()
+    return text_back
+
+if __name__ == '__main__':
+    try:
+        print echo_client(sys.argv[1])
+    except:
```
Comment (cewing) [inline echo_client.py]: A bare except like this is potentially problematic.  There could be many things that go wrong, not just that someone didn't specify a message.  Be specific in what you are handling.  Which exception type would mean the user hadn't specified a message?  Catch that specific exception type only.

File: test_echo_server.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+def test_server_empty():
+    """Test that an empty string works"""
+    text = ""
+    assert echo_client(text) == text
+
+
+def test_server_unicode():
+    """Test that a unicode string works"""
+    text = u"this is a test"
+    assert echo_client(text) == text
```
Comment (cewing) [inline test_echo_server.py:22]: It would be far more meaningful to send a unicode string that contained non-ASCII characters.  That would cause `UnicodeDecodeError` or `UnicodeEncodeError` exceptions to raise for implicit byte/unicode conversions.  This proves nothing specific about the ability of your code to handle unicode strings because Python is implicitly converting them if needed.

File: test_echo_server.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+
+def test_server_long_string():
+    """Test that a string longer than 32 works"""
+    text = u"this is a test of whether a string longer than 32 workds"
+    assert echo_client(text) == text
```
Comment (cewing) [inline test_echo_server.py:38]: Could you add one test that verifies that a string exactly as long as one server buffer works as well?  That helps to catch edge cases where the handling of socket closing is not correct.

**Comment by henry808** [comment]
Thanks for the feedback! I made all changes that you suggested.
